### PR TITLE
Pull Request for CSRK code

### PR DESCRIPTION
**Pull request by Sondar74**
This code is based on the paper "A characterization of energy-preserving methods and the construction of parallel integrators for Hamiltonian systems" (https://doi.org/10.48550/arXiv.1505.02537).

Added a `struct` in BSeries.jl called `ContinuousStageRungeKutta` and a function `CSRK()`.

This code requires to import SymPy in order to work with symbolic variables for the polynomials generation and the integration process. The BSeries.jl was modified by adding a function which generates a polynomial for a square matrix, and another function elementary_differentials_csrk which returns the elementary differential for a given square matrix and a RootedTree. Only this last function is being exported. An exmaple of its functionality can be tested as follows:

```
#define a matrix M
M = [-6//5 72//5 -36 24;
72//5 -144//5 -48 72;
-36 -48 720 -720;
24 72 -720 720]
csrk = CSRK(M)
series = bseries(csrk, 4)
```

### CODE REVIEW

File: src/CSRK.jl [hunk 1]
```diff
@@ -0,0 +1,76 @@
+using BSeries
+using SymPy
+using LinearAlgebra
+using PyCall
+
+# This function generates a polinomial
```
Comment (ketch) [inline src/CSRK.jl:6]: Remember to use English, here and throughout.

File: src/BSeries.jl [hunk 2]
```diff
@@ -1447,4 +1448,437 @@ include("latexify.jl")
     include("precompile.jl")
 end
 
+
```
Comment (ranocha) [inline src/BSeries.jl:1451]: Please clean up this PR. It looks to me like there is a huge amount of changes that is not really related to what you would like to do in this PR - in particular all changes to this file as well as `src/Energy_Preserving.jl`?

File: src/Energy_Preserving.jl [hunk 3]
```diff
@@ -0,0 +1,447 @@
+module Energy_Preserving
```
Comment (ranocha) [inline src/Energy_Preserving.jl:1]: Shouldn't be here?

File: test/runtests.jl [hunk 4]
```diff
@@ -1897,4 +1897,65 @@ using Aqua: Aqua
             Aqua.test_project_toml_formatting(BSeries)
         end
     end
+    #We test the code Energy_Preserving.jl using the function 'OrderMethod(A,b) for a given
```
Comment (ranocha) [inline test/runtests.jl:1900]: Should also be available in `main` in the latest version?

**Review by ranocha — CHANGES_REQUESTED** [review]

**Comment by ranocha** [comment]
If all that is new in this PR is contained in `src/CSRK.jl`, it may be simpler to just create a new PR from a clean base, adding only this single file.
Again, please consider incorporating the new functionality into the package such that it will be available when `using BSeries`

**Comment by Sondar74** [comment]
> If all that is new in this PR is contained in `src/CSRK.jl`, it may be simpler to just create a new PR from a clean base, adding only this single file. Again, please consider incorporating the new functionality into the package such that it will be available when `using BSeries`

You are right. I will make another pull request only with the BSeries.jl file changed

**Comment by ranocha** [comment]
Are you working in this PR or in #141?

**Comment by Sondar74** [comment]
> Are you working in this PR or in #141?

I had problems running the updated code in the other one. Maybe it will be better to keep on working with this one. 
I created the `struct` 'ContinuousRungeKuttaMethod' and a function `CSRK` (an example of its functionality is included in the description of this PR).

**Comment by ranocha** [comment]
Ok. Please close the PR you don't want to continue working on. And please resolve the conflicts in the PR you are working on.

File: Project.toml [hunk 5]
```diff
@@ -1,40 +0,0 @@
-name = "BSeries"
```
Comment (ranocha) [inline Project.toml:1]: You shouldn't delete this file

File: src/BSeries.jl [hunk 6]
```diff
@@ -1447,4 +1482,505 @@ include("latexify.jl")
...
+This function checks whether a method is energy preserving 
+for a given order s. 
+    Inputs: 
+        A: square matrix
+        b: vector of Butcher array
+        s: integer, the order
+    Output:
+        True/False
+"""
+function is_energy_preserving(A,b,s)
```
Comment (ranocha) [inline src/BSeries.jl:1498]: This still looks like a merge conflict

File: test/Project.toml [hunk 7]
```diff
@@ -1,15 +0,0 @@
-[deps]
```
Comment (ranocha) [inline test/Project.toml:1]: You shouldn't delete this file

File: test/runtests.jl [hunk 8]
```diff
@@ -1,1900 +0,0 @@
-using Test
```
Comment (ranocha) [inline test/runtests.jl:1]: You shouldn't delete this file

**Review by ranocha — CHANGES_REQUESTED** [review]

**Pull request closed without merging**